qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -221,9 +255,79 @@ def get_fidelity(
         circuit: QuantumCircuit,
         backend: Optional[Union[Backend, QuantumInstance]] = None,
         expectation: Optional[ExpectationBase] = None,
+        *,
+        sampler: Optional[BaseSampler] = None,
```

Comment:
Is there a reason not to pass Fidelity more directly here. It could still default to ComputeUncompute with a Sampler - or do you see this needing that specific Fidelity? There is also the `fidelity` property in the constructor. Would allowing that to be Fidelity instance too simplify building an instance of this for a user rather than having to more directly deal with functions?

Reply:
Yeah, the idea was to have this function be as close as possible to the original which was just a convenience constructor for the compute-uncompute fidelity. If users start handling fidelity classes directly, I don't this function is much needed since users can just do
```
def fidelity(x, y):
    return MyFidelity().run(circuit, circuit, x, y).result().fidelities

qnspsa = QNSPSA(fidelity, ...)
```
:)

Reply:
Right they can, but it seems to me might be nicer if they could directly pass a Fidelity and we did that wrapping internally like was done for the ComputeUncompute.

Reply:
I'm not so convinced it's worth it to have a function to simplify from
```python
def fidelity(x, y):
    return MyFidelity(Sampler()).run(circuit, circuit, x, y).result().fidelities

qnspsa = QNSPSA(fidelity, ...)
```
to
```python
fidelity = QNSPSA.get_fidelity(circuit, MyFidelity(Sampler())
qnspsa = QNSPSA(fidelity, ...)
```
🤔 

But we can of course discuss and add it later! 🙂